docs_src/design/adr/0005-Service-Self-Config.md

```diff
@@ -0,0 +1,128 @@
+
+If the service finds the top-level (root) namespace is already populated with configuration information it will then read that configuration information from the configuration provider under namespace for that service (and ignore what is in the local configuration file).
+
+If the service finds the top-level (root) namespace is not populated with configuration information, it will read its local configuration file and populate the configuration provider (under the namespace for the service) with configuration read from the local configuration file.
+
+A configuration provider can be specified with a command line argument (the -cp / --configProvider) or environment variable (the EDGEX_CONFIGURATION_PROVIDER environmental variable which overrides the command line argument).
+> NOTE:  the environmental variables are typically uppercase but there have been inconsistencies in environmental variable casing (example:  edgex_registry).  This should be considered and made consistent in a future major release.
+
+**Using the local configuration file**
+
+When a configuration provider _isn't_ specified, the service just uses the configuration in its local configuration file.  That is the service uses the configuration in the file associated with the profile, config filename and config file directory command line options or environmental variables.  In this case, the service does not contact the configuration service (Consul) for any configuration information.
+  
+NOTE:  As the services now self seed and deployment specific changes can be made via environment overrides, it will no longer be necessary to have a Docker profile configuration file in each of the service directories (example:  https://github.com/edgexfoundry/edgex-go/blob/master/cmd/core-data/res/docker/configuration.toml).  See Consequences below.  It will still be possible for users to use the profile mechanism to specify a Docker configuration, but it will no longer be required and not the recommended approach to providing Docker container specific configuration.
+
+### Overrides
+Environmental variables override configuration or command line option values in the following ways
+- Environmental variables override local configuration; that is when configuration for a service is obtained from the local config file
+- Environmental variables override configuration values as they are pushed (self seeded) into the configuration service (Consul).  This override only occurs once (as the values are pushed / seeded into Consul from the service).  Once pushed, configuration values are used from Consul and any environmental value is ignored unless the -o/--overwrite flag is on.  In other words, environmental variables are considered the record of truth when specified for configuration.  The configuration service (Consul) is the record of truth for all other configuration.  The name of the environmental variable must match the path names in Consul.
```

Comment:
The change we decided on Monday was that environment variables used to override configuration always take precedence whether sourced locally or from the config provider/Consul. So suggested wording here is just that:

**## Overrides**
_Environment variables used to override configuration always take precedence whether configuration is being sourced locally or read from the config provider/Consul._

_**Note** - this means that a configuration value that is being overridden by an environment variable will always be the source of truth, even if the same configuration is changed directly in Consul._

I think this replaces everything above except maybe the line about the name of the environment variable having to match the path name in Consul.

@lenny-intel does that sound right to you?

Reply:
Sounds good.